protobuf/grpc_service.proto

```diff
@@ -1477,3 +1486,74 @@ message CudaSharedMemoryUnregisterRequest
 //@@   Response message for CudaSharedMemoryUnregister.
 //@@
 message CudaSharedMemoryUnregisterResponse {}
+
+//@@
+//@@.. cpp:var:: message TraceRequest
+//@@
+//@@   Request message for Trace.
+//@@
+message TraceRequest
+{
+  //@@
+  //@@  .. cpp:var:: message SettingValue
+  //@@
+  //@@     The values to be associated with a trace setting.
+  //@@
+  message SettingValue
+  {
+    //@@
+    //@@    .. cpp:var:: string value (repeated)
+    //@@
+    //@@       The value.
+    //@@
+    repeated string value = 3;
+  }
+
+  //@@  .. cpp:var:: map<string,SettingValue> settings
+  //@@
+  //@@     The new setting values to be updated,
+  //@@     settings that are not specified will remain unchanged.
+  //@@
+  map<string, SettingValue> settings = 1;
+
+  //@@
+  //@@  .. cpp:var:: string name
+  //@@
+  //@@     The name of the model to apply the new trace settings.
+  //@@     If not given, the new settings will be applied globally.
+  //@@
+  string model_name = 2;
+
+  //@@
+  //@@  .. cpp:var:: bool clear_setting
+  //@@
+  //@@     Clear the current model specific setting if specified, and
+  //@@     any new setting values will not be applied.
+  //@@     This option will be ignored if 'model_name' is not given.
+  //@@
+  bool clear_setting = 3;
+}
+
+message TraceResponse
```

Comment:
Which settings are returned if the request didn't set a model_name? I guess because the setting is applied globally same set of settings will be returned?
Shouldn't  there a way from client side just to retrieve trace settings for all models without updating them?

Reply:
If model name is not specified, global setting will be returned. I haven't considered the case where the user wants to get the setting for all models, so it is just either returning the global or model.

If the field is not specified in the request, then it won't update the field. So user can just send a request without `settings` to only retrieve the setting